File: 7. Reverse Integer/ReverseInteger.cpp
class Solution {
public:
    int reverse(int x) {
          long long int newN =0, left =0;
          while(x != 0)
          {  
                 left = x%10;  
                 newN = newN*10 + left;  
                 x = x/10;  
          }  
        if((long long int)((int)newN)==newN)
            return newN;
        else
            return 0;
    }
};

File: 7. Reverse Integer/ReverseInteger.java
public class Solution {
    public int reverse(int x) {
        long num = 0;
        boolean neg = (x < 0);
        long xx = neg? -x: x;
        while(xx != 0)
        {
            num = num * 10 + xx % 10;
            xx /= 10;
        }
        if(neg)
        {
            num = -num;
        }
        if(((long) ((int) num)) == num)
        {
            return (int) num;
        }
        else
        {
            return 0;
        }
    }
}